packages/marketplace/contracts/RoyaltiesRegistry.sol

```diff
@@ -200,40 +200,60 @@ contract RoyaltiesRegistry is OwnableUpgradeable, IRoyaltiesProvider {
-                bool result
-            ) {
-                if (result) {
-                    try IMultiRoyaltyRecipients(token).getRecipients(tokenId) returns (
-                        Recipient[] memory multiRecipients
-                    ) {
-                        uint256 multiRecipientsLength = multiRecipients.length;
-                        royalties = new Part[](multiRecipientsLength);
-                        uint256 sum = 0;
-                        for (uint256 i; i < multiRecipientsLength; i++) {
-                            Recipient memory splitRecipient = multiRecipients[i];
-                            royalties[i].account = splitRecipient.recipient;
-                            uint256 splitAmount = (splitRecipient.bps * royaltyAmount) / WEIGHT_VALUE;
-                            royalties[i].value = splitAmount;
-                            sum += splitAmount;
-                        }
-                        // sum can be less than amount, otherwise small-value listings can break
-                        require(sum <= royaltyAmount, "RoyaltiesRegistry: Invalid split");
-                        return royalties;
-                    } catch {
-                        return _calculateRoyalties(receiver, royaltyAmount);
-                    }
-                } else {
-                    return _calculateRoyalties(receiver, royaltyAmount);
-                }
-            } catch {
+            if (_checkGetRecipientsInterface(token)) {
+                royalties = _getRecipients(token, tokenId, receiver, royaltyAmount);
+            } else {
                 return _calculateRoyalties(receiver, royaltyAmount);
             }
         } catch {
             return new Part[](0);
         }
     }
 
+    /// @notice try to get the royalties recipients and do the royalties calculations
+    /// @param token address of token
+    /// @param tokenId Id of the token
+    /// @param receiver address of the royalty receiver
+    /// @param royaltyAmount royalty amount to be paid
+    /// @return royalties 2981 royalty array
+    function _getRecipients(
+        address token,
+        uint256 tokenId,
+        address receiver,
+        uint256 royaltyAmount
+    ) internal view returns (Part[] memory royalties) {
+        try IMultiRoyaltyRecipients(token).getRecipients(tokenId) returns (Recipient[] memory multiRecipients) {
+            uint256 multiRecipientsLength = multiRecipients.length;
+            royalties = new Part[](multiRecipientsLength);
+            uint256 sum = 0;
+            for (uint256 i; i < multiRecipientsLength; i++) {
+                Recipient memory splitRecipient = multiRecipients[i];
+                royalties[i].account = splitRecipient.recipient;
+                uint256 splitAmount = (splitRecipient.bps * royaltyAmount) / WEIGHT_VALUE;
+                royalties[i].value = uint96(splitAmount);
+                sum += splitAmount;
+            }
```

Comment:
have all the calculations of royalties in only one place if possible.
this looks like a lot like _calculateRoyalties

Reply:
Created a new branch for that:
https://github.com/thesandboxgame/sandbox-smart-contracts/tree/task/merge_royalties_calculation

Reply:
Draft-PR -> https://github.com/thesandboxgame/sandbox-smart-contracts/pull/1247